test/powershell/Language/Scripting/ErrorHistoryId.Tests.ps1

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe "Error HistoryId Tests" -Tags "CI" {
+
+    BeforeAll {
+        $setting = [System.Management.Automation.PSInvocationSettings]::New()
+        $setting.AddToHistory = $true
+
+        function RunCommand($ps, $command) {
+            $ps.Commands.Clear()
+            $ps.AddScript($command).Invoke($null, $setting)
+        }
+    }
+
+    BeforeEach {
+        $ps = [PowerShell]::Create("NewRunspace")
+    }
+
+    AfterEach {
+        $ps.Dispose()
+    }
+
+    It "Error from 'Write-Error' has the right history Id" {
+        $null = RunCommand $ps "function bar { Write-Error 'abc' }"   ## 1st in history
+        $null = RunCommand $ps "bar"                                  ## 2nd in history
+
+        $ps.HadErrors | Should -BeTrue
+        RunCommand $ps '$Error[0].InvocationInfo.HistoryId' | Should -Be 2
+    }
+
+    It "Error from 'PSCmdlet.WriteError' has the right history Id" {
+        ## 1st in history
+        $null = RunCommand $ps @'
+function bar {
+    [CmdletBinding()]
+    param()
+
+    $er = [System.Management.Automation.ErrorRecord]::new(
+        [System.ArgumentException]::new(),
+        'PSCmdlet.WriteError',
+        [System.Management.Automation.ErrorCategory]::InvalidOperation,
+        $null)
+    $PSCmdlet.WriteError($er)
+}
+'@
+        ## 2nd in history
+        $null = RunCommand $ps 'bar'
+
+        $ps.HadErrors | Should -BeTrue
+        $err = RunCommand $ps '$Error[0]'
+        $err.FullyQualifiedErrorId | Should -BeExactly "PSCmdlet.WriteError,bar"
+        $err.InvocationInfo.HistoryId | Should -Be 2
+    }
+
+    It "Error from 'PSCmdlet.ThrowTerminatingError' has the right history Id" {
+        ## 1st in history
+        $null = RunCommand $ps @'
+function bar {
+    [CmdletBinding()]
+    param()
+
+    $er = [System.Management.Automation.ErrorRecord]::new(
+        [System.ArgumentException]::new(),
+        'PSCmdlet.ThrowTerminatingError',
+        [System.Management.Automation.ErrorCategory]::InvalidOperation,
+        $null)
+    $PSCmdlet.ThrowTerminatingError($er)
+}
+'@
+        ## 2nd in history
+        $null = RunCommand $ps 'bar'
+        $ps.HadErrors | Should -BeTrue
+        $err = RunCommand $ps '$Error[0]'
+        $err.FullyQualifiedErrorId | Should -BeExactly "PSCmdlet.ThrowTerminatingError,bar"
+        $err.InvocationInfo.HistoryId | Should -Be 2
+    }
+
+    It "Error from 'Throw' has the right history Id - 1" {
```

Comment:
This test and the next one are very similar, you should turn it into `-TestCases`, something like:

```powershell
It "Error from 'Throw' has the right history Id: <commandOne>" -TestCases @(
  @{ commandOne = '1+1'; CommandTwo = "throw 'abc'" }
  @{ commandOne = "function bar { throw 'abc' }"; commandTwo = 'bar' }
){
        param ($commandOne, $commandTwo)
        try {
            ## 1st in history
            $null = RunCommand $ps $commandOne
            ## 2nd in history
            $null = RunCommand $ps $commandTwo
        } catch {
            ## ignore the exception
        }

        $ps.HadErrors | Should -BeTrue
        $err = RunCommand $ps '$Error[0]'
        $err.FullyQualifiedErrorId | Should -BeExactly 'abc'
        $err.InvocationInfo.HistoryId | Should -Be 2
    }
```

Reply:
Updated using `-TestCases`.